### PR TITLE
BUG: Fisher test corner case/gh #276

### PR DESCRIPTION
**Pull request by kshedden**
This should resolve the errors discussed in #276.

### CODE REVIEW

File: src/fisher.jl [hunk 1]
```diff
@@ -137,7 +137,9 @@ end
 
 function pvalue_both_minlike(x::FisherExactTest, ω::Float64=1.0)
     a, b, c, d = reorder(x.a, x.b, x.c, x.d)
-
+    if a == c == 0 || b == d == 0
+        return 1
```
Comment (ararslan) [inline src/fisher.jl:141]:
```suggestion
        return 1.0
```
For type stability. Looking through the code, I think everything for Fisher's exact test ends up as `Float64`, so we don't need to do any shenanigans to figure out the appropriate return type.

File: test/fisher.jl [hunk 2]
```diff
@@ -104,6 +104,24 @@ t = HypothesisTests.FisherExactTest(4, 1, 20, 1)
...
+# Corner cases gh #276
+t = HypothesisTests.FisherExactTest(5, 0, 5, 0)
+@test pvalue(t; tail=:left) ≈ 1
+@test pvalue(t; tail=:right) ≈ 1
+@test pvalue(t; method=:central) ≈ 1
+@test pvalue(t; method=:minlike) ≈ 1
+@test_ci_approx confint(t; tail=:left) (0.0, Inf)
+@test_ci_approx confint(t; tail=:right) (0.0, Inf)
+@test_ci_approx confint(t; method=:central) (0.0, Inf)
+t = HypothesisTests.FisherExactTest(0, 5, 0, 5)
```
Comment (nalimilan) [inline test/fisher.jl:116]:
```suggestion

t = HypothesisTests.FisherExactTest(0, 5, 0, 5)
```

**Review by nalimilan — APPROVED** [review]
Thanks!

**Review by ararslan — APPROVED** [review]
Great, thanks for the contribution!
